addons/web/static/src/legacy/js/fields/basic_fields.js

```diff
@@ -76,6 +76,42 @@ var TranslatableFieldMixin = {
 
 var DynamicPlaceholderFieldMixin = {
     DYNAMIC_PLACEHOLDER_TRIGGER_KEY: '#',
+    /**
+     * Overridable method that ensure the modelSelectorPopover is
+     * inserted at the right place in the DOM and positioned properly.
+     *
+     * @public
+     * @param {ModelFieldSelectorPopover} modelSelector     *
+     * @returns {Promise<ModelFieldSelectorPopover>}
+     */
+    insertAndPositionModelSelector: async function (modelSelector) {
+        const $insertAfterElement = this.$el.first();
+        const $modalParent = $insertAfterElement.parents('div.modal-content');
+        let relativeParent = $modalParent[0];
+        if (!relativeParent) {
+            relativeParent = $insertAfterElement[0];
+            while(!relativeParent || !['absolute', 'relative'].includes(getComputedStyle(relativeParent).position)) {
+                relativeParent = relativeParent.offsetParent;
+            }
+        }
+
+        const relatedElementPosition = $insertAfterElement[0].getBoundingClientRect();
+        const relativeParentPosition = relativeParent.getBoundingClientRect();
+
+        let topPosition = relatedElementPosition.top + relatedElementPosition.height - relativeParentPosition.top
+        let leftPosition = relatedElementPosition.left - relativeParentPosition.left;
+        // If we are inside a modal environment,
+        // we need to append the ModelFieldSelectorPopover outside the
+        // modal body, to be sure the overflow will be visible.
+        if ($modalParent.length) {
+            await modelSelector.appendTo($modalParent);
+        } else {
+            await modelSelector.insertAfter($insertAfterElement);
+        }
```

Comment:
Could we move the insertion part outside of the method ? It doesn't seem to be needed in the override. That way we can call the method `_positionModelSelection`.